databaseGenerator.js

```diff
@@ -2,141 +2,157 @@ const fs = require('fs');
   }
   sql = sql.join('\n');
   return sql;
-}
-var reviews = [];
-for (var i = 1; i <= numReviews; i++) {
+};
+let reviews = [];
+for (let i = 1; i <= numReviews; i += 1) {
   reviews.push(createReview(i));
 }
-//========================================================
+
 // PHOTOS SECTION
-var createPhoto = function(index) {
-  var photo = {};
+const avatarIds = [];
+const photoIds = [];
+photoToUserIds.forEach((ele) => {
+  avatarIds.push(ele[1]);
+  photoIds.push(ele[0]);
+});
+module.exports.photoIds = photoIds;
+const createPhoto = (index) => {
+  const photo = {};
   photo.id = index;
-  if (Math.random().toFixed(2) > .65) {
+  const i = photoIds.indexOf(index);
+  if (i > -1) {
+    photo.user_id = avatarIds[i];
+    photo.business_id = 'NULL';
+  } else {
     photo.business_id = businesses[Math.floor(Math.random() * numBusinesses)].id;
+    photo.user_id = users[Math.floor(Math.random() * numUsers)].id;
   }
-  photo.user_id = users[Math.floor(Math.random() * numUsers)].id;
   return photo;
-}
-var convertPhotosToSql = function(photos) {
-  var sql = [];
-  for (var i = 0; i < photos.length; i++) {
-    sql[i] = `INSERT INTO photos (ID, BUSINESS_ID, USER_ID) VALUES ("${photos[i].id}", ${photos[i].business_id ? `"` + photos[i].business_id + `"` : "NULL"}, ${photos[i].user_id ? `"` + photos[i].user_id + `"` : "NULL"});`;
+};
+const convertPhotosToSql = (photos) => {
+  let sql = [];
+  for (let i = 0; i < photos.length; i += 1) {
+    sql[i] = `INSERT INTO photos (ID, BUSINESS_ID, USER_ID) 
+    VALUES ("${photos[i].id}", ${photos[i].business_id === "NULL" ? null : ['"', photos[i].business_id, '"'].join('') }, "${photos[i].user_id}");`;
   }
   sql = sql.join('\n');
   return sql;
-}
-var photos = [];
-for (var i = 1; i <= numPhotos; i++) {
+};
+let photos = [];
+for (let i = 1; i <= numPhotos; i += 1) {
   photos.push(createPhoto(i));
 }
-//========================================================
+
 // MESSAGES SECTION
-var createMessage = function(index) {
-  var message = {};
+const createMessage = (index) => {
+  const message = {};
   message.id = index;
   message.author_id = users[Math.floor(Math.random() * numUsers)].id;
   message.recipient_id = users[Math.floor(Math.random() * numUsers)].id;
-  message.messageText = faker.lorem.paragraph();
+  message.messageText = faker.lorem.paragraph(1);
   message.subject = faker.lorem.sentence();
   return message;
-}
-var convertMessagesToSql = function(messages) {
-  var sql = [];
-  for (var i = 0; i < messages.length; i++) {
+};
+const convertMessagesToSql = (messages) => {
+  let sql = [];
+  for (let i = 0; i < messages.length; i += 1) {
     sql[i] = `INSERT INTO messages (ID, AUTHOR_ID, RECIPIENT_ID, SUBJECT, MESSAGETEXT) VALUES ("${messages[i].id}", "${messages[i].author_id}", "${messages[i].recipient_id}", "${messages[i].subject}", "${messages[i].messageText}");`;
   }
   sql = sql.join('\n');
```

Comment:
I like this as a method for inserting the data into the DB. One insert with all the data.
I think in my case I'm doing an individual insert for every entry. I might refactor to something like this later.